sklearn/manifold/isomap.py

```diff
@@ -96,11 +149,51 @@ def __init__(self, n_neighbors=5, n_components=2, eigen_solver='auto',
         self.neighbors_algorithm = neighbors_algorithm
         self.nbrs_ = NearestNeighbors(n_neighbors=n_neighbors,
                                       algorithm=neighbors_algorithm)
+        self.n_landmarks = n_landmarks
+        self.landmarks_ = landmarks
```

Comment:
The ``init`` function should store each named parameter by name. This is so that the ``clone()`` functionality will work correctly.